pkg/discovery/servicehandler.go

```diff
@@ -316,92 +263,43 @@ func doesAPIContainAnyMatchingTag(tags, apiTags []string) bool {
 	return false
 }
 
-// Helper function to remove existing back-end security policies
-func removeOASpolicies(specContent []byte, oasVersion string) ([]byte, error) {
-	spec := make(map[string]interface{})
-	err := json.Unmarshal(specContent, &spec)
-	if err != nil {
-		return specContent, err
-	}
-	// Deleting any pre-existing security definitions
-	if oasVersion == "v2" {
-		if spec["securityDefinitions"] != nil {
-			delete(spec, "securityDefinitions")
-		}
-	}
-	if oasVersion == "v3" {
-
-		oas3Spec := openapi3.Swagger{}
-		err := json.Unmarshal(specContent, &oas3Spec)
-		if err != nil {
-			return specContent, err
-		}
-
-		// reset to empty
-		oas3Spec.Components.SecuritySchemes = nil
-		return json.Marshal(oas3Spec)
-
-	}
-
-	return json.Marshal(spec)
-}
-
 // TODO improve if fields are not complete, introduce per method swagger definitions, set up logic for multiple auth policies, use policy configurations
-func setOAS2policies(sc []byte, authPolicy string) ([]byte, error) {
+func setOAS2policies(swagger *apic.Oas2Swagger, authPolicy string) ([]byte, error) {
 	// Removing pre-existing auth security policies
-	sc, err := removeOASpolicies(sc, "v2")
-	if err != nil {
-		return sc, err
-	}
-
-	oas2Spec := openapi2.Swagger{}
-	err = json.Unmarshal(sc, &oas2Spec)
-	if err != nil {
-		return nil, err
-	}
-
+	swagger.SecurityDefinitions = make(map[string]*openapi2.SecurityScheme)
 	switch authPolicy {
 	case apic.Apikey:
-		ssp := openapi2.SecuritySchemeProps{
+		ss := openapi2.SecurityScheme{
```

Comment:
Switched from the go-openapi library to the kin/openapi2 library

Reply:
Might need to validate this does not cause conflict or a breaking change.